homeassistant/components/mqtt_statestream/__init__.py

```diff
@@ -38,15 +50,23 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the MQTT state feed."""
-    conf = config[DOMAIN]
+    callback_handler: CALLBACK_TYPE
+
+    conf: ConfigType = config[DOMAIN]
     publish_filter = convert_include_exclude_filter(conf)
-    base_topic = conf.get(CONF_BASE_TOPIC)
-    publish_attributes = conf.get(CONF_PUBLISH_ATTRIBUTES)
-    publish_timestamps = conf.get(CONF_PUBLISH_TIMESTAMPS)
+    base_topic: str = conf[CONF_BASE_TOPIC]
+    publish_attributes: bool = conf[CONF_PUBLISH_ATTRIBUTES]
+    publish_timestamps: bool = conf[CONF_PUBLISH_TIMESTAMPS]
+    ha_started = asyncio.Event()
     if not base_topic.endswith("/"):
         base_topic = f"{base_topic}/"
 
-    async def _state_publisher(entity_id, old_state, new_state):
+    async def _state_publisher(
+        entity_id: str, old_state: State | None, new_state: State | None
+    ) -> None:
+        if not ha_started.is_set():
+            return
+
         if new_state is None:
```

Comment:
Move this + `publish_filter` to an `event_filter` passed to `hass.states.async_listen`